core/src/main/scala/com/nvidia/spark/rapids/tool/qualification/QualOutputWriter.scala

```diff
@@ -255,17 +255,50 @@ class QualOutputWriter(outputDir: String, reportReadSchema: Boolean,
     }
   }
 
-  def writeExecReport(sums: Seq[QualificationSummaryInfo], order: String) : Unit = {
+  def writeExecReport(sums: Seq[QualificationSummaryInfo]) : Unit = {
     val csvFileWriter = new ToolTextFileWriter(outputDir,
       s"${QualOutputWriter.LOGFILE_NAME}_execs.csv",
       "Plan Exec Info", hadoopConf)
     try {
-      val headersAndSizes = QualOutputWriter.getDetailedExecsHeaderStrings
       csvFileWriter.write(QualOutputWriter.constructDetailedHeader(
         QualOutputWriter.getDetailedExecsHeaderStrings, ",", false))
+      val delimiter = ","
+      val booleanTrue = "true"
+      val booleanFalse = "false"
+      val zeroDurationStr = "0"
       sums.foreach { sumInfo =>
-        val appRows = QualOutputWriter.constructExecsInfo(sumInfo, headersAndSizes, ",", false)
-        appRows.foreach(csvFileWriter.write(_))
+        val appIDCSVStr = StringUtils.reformatCSVString(sumInfo.appId)
+        sumInfo.planInfo.foreach { pInfo =>
+          val execInfos = pInfo.execInfo.flatMap { eInfo =>
+            eInfo.children.getOrElse(Seq.empty) :+ eInfo
+          }.sortBy(eInfo => eInfo.nodeId)
+          val sqlIDStr = pInfo.sqlID.toString
+          val rows = execInfos.map { info =>
+            val (childrenExecsStr, nodeIdsStr) = if (info.children.isDefined) {
+              (StringUtils.reformatCSVString(
+                info.children.get.map(_.exec).mkString(":")),
+                StringUtils.reformatCSVString(
+                  info.children.get.map(_.nodeId).mkString(":")))
+            } else {
+              ("", "")
+            }
+            Seq(appIDCSVStr, sqlIDStr, StringUtils.reformatCSVString(info.exec),
+              StringUtils.reformatCSVString(info.expr),
+              if (info.duration.isDefined) info.duration.get.toString else zeroDurationStr,
+              info.nodeId.toString,
+              if (info.isSupported) booleanTrue else booleanFalse,
+              StringUtils.reformatCSVString(info.stages.mkString(":")),
+              childrenExecsStr,
+              nodeIdsStr,
+              if (info.shouldRemove) booleanTrue else booleanFalse,
```

Comment:
Question: Why do we require the `booleanTrue` and `booleanFalse` variables instead of directly using `info.shouldRemove.toString`?

Reply:
Good question @parthosa !
I assume that `info.shouldRemove.toString` is going to create a new thread each time it is called. We can actually test that by checking the address of the string returned from the call in case Scala is optimized to return a string from an internal pool.
Using the booleanTrue allocates the object only once and shares it with all other records.